mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraSchemaVersionManager.java

```diff
@@ -0,0 +1,103 @@
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.OK;
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.OLD_BUT_SUPPORTED;
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.TOO_OLD;
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.TOO_RECENT;
+
+public class CassandraSchemaVersionManager {
+    public static final int MIN_VERSION = 1;
+    public static final int MAX_VERSION = 1;
+
+    public final int minVersion;
+    public final int maxVersion;
+    public final int version;
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(CassandraSchemaVersionManager.class);
+
+
+    public CassandraSchemaVersionManager(CassandraSchemaVersionDAO schemaVersionDAO) {
+        this(schemaVersionDAO, MIN_VERSION, MAX_VERSION);
+    }
+
+    @Inject
+    @VisibleForTesting
+    CassandraSchemaVersionManager(CassandraSchemaVersionDAO schemaVersionDAO, int minVersion, int maxVersion) {
+        this.minVersion = minVersion;
+        this.maxVersion = maxVersion;
+        this.version = schemaVersionDAO.getCurrentSchemaVersion().join();
+    }
+
+    public enum SchemaVersionState {
+        TOO_OLD, OLD_BUT_SUPPORTED, OK, TOO_RECENT;
+    }
+
+    public int getVersion() {
+        return version;
+    }
+
+    public SchemaVersionState getSchemaVersionState() {
+        if (version < minVersion) {
+            return TOO_OLD;
+        }
+        else if (version < maxVersion) {
+            return OLD_BUT_SUPPORTED;
+        }
+        else if (version == maxVersion){
+            return OK;
+        }
+        else {
+            return TOO_RECENT;
+        }
+    }
+
+    public void ensureSchemaIsSupported() {
+        SchemaVersionState state = getSchemaVersionState();
+
+        switch (state) {
```

Comment:
We should not enforce some hypothetic future expected changes in today code at the detriment of today code quality. Today we just need to report, or crash. Then this "state" is not needed, and is just redoundant.